client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-patterns-map-by-category.ts

```diff
@@ -3,17 +3,20 @@ import type { Pattern, Category } from '../types';
 
 const usePatternsMapByCategory = ( patterns: Pattern[], categories: Category[] ) => {
 	return useMemo( () => {
-		const categoriesMap = categories.reduce(
-			( map, category ) => ( {
-				...map,
-				[ category.name ]: [],
-			} ),
-			{}
-		) as { [ key: string ]: Pattern[] };
+		const categoriesMap = categories.reduce( ( map, category ) => {
+			return category.name
+				? {
+						...map,
+						[ category.name ]: [],
+				  }
+				: map;
+		}, {} ) as { [ key: string ]: Pattern[] };
 
 		patterns.forEach( ( pattern ) => {
-			pattern.categories.forEach( ( category ) => {
-				categoriesMap[ category ]?.push( pattern );
+			Object.values( pattern.categories ).forEach( ( { slug } ) => {
+				if ( slug ) {
+					categoriesMap[ slug ]?.push( pattern );
+				}
 			} );
 		} );
 		return categoriesMap;
```

Comment:
I might understand it wrong, but I think we could create `categoriesMap` in one loop (without `categories.reduce`).

```ts
		const categoriesMap = {}

		patterns.forEach( ( pattern ) => {
			Object.values( pattern.categories ).forEach( ( { slug } ) => {
				if ( ! slug ) {
					return;
				}
				if ( ! categoriesMap[ slug ] ) {
					categoriesMap[ slug ] = []
				}
				categoriesMap[ slug ].push( pattern );
			} );
		} );

		return categoriesMap;
```

Reply:
Right, thanks :)